File: IntroduccionPOO/Ejercicio2/Coche.java

package introduccionpoo;

public class Coche {
    private String marca;
    private String modelo;
    private int velocidad;

    public Coche(String marca, String modelo) {
        this.marca = marca;
        this.modelo = modelo;
        this.velocidad = 0;
    }

    public void acelerar() {
        velocidad += 10;
    }

    public void frenar() {
        velocidad -= 5;
        if (velocidad < 0) velocidad = 0;
    }

    public void mostrarVelocidad() {
        System.out.println(marca + " " + modelo + " va a " + velocidad + " km/h");
        System.out.println("noemi chino blanco");
    }

   
    public static void main(String[] args) {
        Coche coche1 = new Coche("Toyota", "Corolla");
        Coche coche2 = new Coche("Ford", "Mustang");

        coche1.acelerar();
        coche1.acelerar();
        coche1.frenar();
        coche1.mostrarVelocidad();

        coche2.acelerar();
        coche2.frenar();
        coche2.frenar();
        coche2.mostrarVelocidad();
    }
}

File: IntroduccionPOO/Ejercicio2/coche.py
class Coche:
    def __init__(self, marca, modelo):
        self.marca = marca
        self.modelo = modelo
        self.velocidad = 0

    def acelerar(self):
        self.velocidad += 10

    def frenar(self):
        self.velocidad -= 5
        if self.velocidad < 0:
            self.velocidad = 0

    def mostrar_velocidad(self):
        print(f"{self.marca} {self.modelo} va a {self.velocidad} km/h")

coche1 = Coche("Toyota", "ferr")
coche2 = Coche("Ferrari", "Mustang")

coche1.acelerar()
coche1.acelerar()
coche1.frenar()
coche1.mostrar_velocidad()

coche2.acelerar()
coche2.frenar()
coche2.frenar()
coche2.mostrar_velocidad()
